x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -235,22 +237,44 @@ protected PhysicalPlan rule(FilterExec filterExec, LocalPhysicalOptimizerContext
 
         public static boolean canPushToSource(Expression exp, Predicate<FieldAttribute> hasIdenticalDelegate) {
             if (exp instanceof BinaryComparison bc) {
-                return isAttributePushable(bc.left(), bc, hasIdenticalDelegate) && bc.right().foldable();
+                Expression left = bc.left();
+                Expression right = bc.right();
+
+                /**
+                 * Because of {@link org.elasticsearch.xpack.esql.type.EsqlDataTypes#widenSmallNumericTypes(DataType)} we should not
+                 * encounter some data types - we include them anyway to be on the safe side.
+                 */
```

Comment:
I'll clarify the comment (and yes, it should be C-style, oof). What I meant is that the check covers some data types that currently cannot occur on the left hand side (shorts and bytes just become integers in terms of data types due to widening).

I think we should leave the out-of-range check in place; otherwise, this rule will be incorrect unless it's always run after `OutOfRangeBinaryComparison`. This avoids unnecessary dependencies between optimization rules - especially since `OutOfRangeBinaryComparison` happens at a completely different step.

Reply:
Ended up just removing the changes to the local planner altogether.